tests/run.sh

```diff
@@ -1,93 +1,106 @@
-
-  if [ $GUI = false ]; then
-    id=$(
-      docker run -d \
-        -e SDL_VIDEODRIVER="dummy" \
-        -p "9001:9001" \
-        -p "21326:21326" \
-        -p "21325:21326" \
-        registry.gitlab.com/satoshilabs/trezor/trezor-user-env/trezor-user-env \
-        "/trezor-user-env/run.sh"
-    )
-    echo "Running docker container with an ID:"
-    echo "$id"
-
-  else
-    xhost +
-    id=$(
-      docker run -d \
-        -e DISPLAY=:0 \
-        --network host \
-        -p "9001:9001" \
-        -p "21326:21326" \
-        -p "21325:21326" \
-        registry.gitlab.com/satoshilabs/trezor/trezor-user-env/trezor-user-env \
-        "/trezor-user-env/run.sh"
-    )
-
-    echo "Running docker container with a GUI support with ID:"
-    echo "$id"
-  fi
+  docker pull "$USER_ENV_IMAGE"
+  dockerID=$(
+    docker run -d \
+      -e SDL_VIDEODRIVER="dummy" \
+      -p "9001:9001" \
+      -p "21326:21326" \
+      -p "21325:21326" \
+      "$USER_ENV_IMAGE"
+  )
+  docker logs -f "$dockerID" &
+  echo "Running docker container with ID $dockerID"
+}
+
+waitForEnv() {
+  echo "Waiting for trezor-user-env to load up..."
+  counter=0
+  max_attempts=60
 
-  echo "Waiting for the trezor-user-env to load up"
-  while ! netstat -tna | grep 'LISTEN\>' | grep -q ':9001\>'; do
-    echo "Waiting"
+  # there is no official support for websockets in curl
+  # trezor-user-env websocket server will recognize request with "close" header and close the connection
+  # otherwise curl will hang in streaming state
+  until (
+    curl -i -s -f \
+      -H "Connection: close" \
+      http://localhost:9001
+  ); do
+    if [ ${counter} -eq ${max_attempts} ]; then
+      echo "trezor-user-env is not running. exiting"
+      exit 1
+    fi
+    counter=$(($counter+1))
+    printf "."
     sleep 1
   done
-  echo "trezor-user-env loaded up"
 
-  echo "Running yarn"
-  yarn jest --config jest.config.integration.js --verbose --detectOpenHandles --forceExit --coverage $COVERAGE
+  echo "trezor-user-env loaded up"
 }
 
 show_usage() {
   echo "Usage: run [OPTIONS] [ARGS]"
   echo ""
   echo "Options:"
-  echo "  -g       Run tests with emulator graphical output"
-  echo "  -f       Use specific firmware version, for example: 2.1.4., 2.3.0"
-  echo "  -i       Included methods only, for example: applySettings,signTransaction"
-  echo "  -e       All methods except excluded, for example: applySettings,signTransaction"
-  echo "  -c       Collect coverage"
+  echo "  -d       Disable docker. Useful when running own instance of trezor-user-env. default: enabled"
+  echo "  -c       Disable backend cache. default: enabled"
+  echo "  -f       Use specific firmware version, example: 2.1.4, 1.8.0 default: 2-master"
+  echo "  -i       Included methods only, example: applySettings,signTransaction"
+  echo "  -e       All methods except excluded, example: applySettings,signTransaction"
+  echo "  -s       actual test script. default: 'yarn test:integration'"
```

Comment:
nitpick: I really like if the options are ordered alphabetically it's then easy to find what you're looking for at a glance.
```suggestion
  echo "  -c       Disable backend cache. default: enabled"
  echo "  -d       Disable docker. Useful when running own instance of trezor-user-env. default: enabled"
  echo "  -e       All methods except excluded, example: applySettings,signTransaction"
  echo "  -f       Use specific firmware version, example: 2.1.4, 1.8.0 default: 2-master"
  echo "  -i       Included methods only, example: applySettings,signTransaction"
  echo "  -s       actual test script. default: 'yarn test:integration'"
```

Reply:
468c833

Reply:
🦊